scripts/hud/df-jump.js

```diff
@@ -20,8 +20,18 @@ class DFJump {
 	static totalLabel = $('#JumpTotalLabel');
 
 	static onLoad() {
-		this.initializeSettings();
-		this.colorClass = COLOR_CLASS.GROUND;
+		if (GameModeAPI.GetCurrentGameMode() === GameMode.DEFRAG) {
+			this.updateHandle = $.RegisterEventHandler(
+				'DFJumpDataUpdate',
+				this.container,
+				this.onDFJumpUpdate.bind(this)
+			);
+			this.initializeSettings();
+			this.colorClass = COLOR_CLASS.GROUND;
+		} else if (this.updateHandle) {
+			$.UnregisterEventHandler('DFJumpDataUpdate', this.container, this.updateHandle);
+			this.updateHandle = null;
+		}
```

Comment:
Could use my crazy `$.RegisterHudPanelForGamemodes` function instead of this, in constructor do
```ts
RegisterHUDPanelForGamemode({
  gamemodes: [GameMode.DEFRAG],
  context: this,
  onLoad: this.onLoad,
  contextPanel: this.container,
  handledEvents: [{event: 'DFJumpDataUpdate', callback: this.onDfJumpData, contextPanel: this.container }]
```
(though as I was reading docs for this, I realised `contextPanel` is a bad name, should be `targetPanel` or something).

Idk, maybe my system is trying to be a bit too clever. But I like keeping all the event registration stuff and handle separate from the rest of the class, and have the onLoad method just deal with init the panel for gamemodes it's actually enabled for.

Reply:
took me a bit to figure out how to implement this but I've updated the code to use your `RegisterHudPanelForGamemodes` function. I agree, it's best to leave this all out of `onLoad`

Reply:
Cheers, sorry if this system is a bit verbose - was there anything in particular that was unclear? Could improve docs, maybe simplify stuff a bit.